pystac/asset.py

```diff
@@ -0,0 +1,244 @@
+from copy import copy
+
+from pystac.stac_object import STACObject
+from pystac import STACError
+from pystac.utils import (is_absolute_href, make_absolute_href, make_relative_href)
+
+
+class STACObjectWithAssets(STACObject):
```

Comment:
Is there a reason to use this type of inheritance instead of a Mixin, similar to the pattern that [LinkMixin](https://github.com/stac-utils/pystac/blob/develop/pystac/stac_object.py#L94) uses?

Reply:
Good catch. I will change it to that approach

Reply:
done